### PR TITLE
EL-3102 - Angular 6 Upgrade

### PR DESCRIPTION
**Pull request by ashley-hunter**
- Updated Documentation site to use Angular 6
- Updated TypeScript to 2.7.2 and fixed some new type errors it found
- Updated Webpack to v4 and updated webpack configs to be similar to the configs used in the Angular CLI (should now notice a big improvement in incremental build times).
- Did a little bit of refactoring, our app module was huge, with loads of imports and declarations, now split it up into child modules should improve code splitting and page load times.
- Removed all AngularJS and Jquery imports - new Angular webpack plugin to properly handle global scripts
- We had loads of modules importing TabsModule incorrectly, they were not using forRoot() so that has been corrected.
- Updated Plunker to use latest versions

### CODE REVIEW

File: docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html [hunk 1]
```diff
@@ -157,6 +157,6 @@ <h4 class="notification-title">Messages Received</h4>
 </blockquote>
 
 <blockquote>
-    <p><strong>Note</strong>: To display web animations in IE 11 and Edge the <code>web-animations-js</code> package must be imported.</p>
+    <p><strong>Note</strong>: To display web animations in IE 11 and Edge the <code>web-animations-js</code> package must be imported (Angular 5 only).</p>
```
[the remark this answers is not in the document]
Comment (ashley-hunter) [reply docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html]: Angular 6 now falls back to CSS animations automatically when web animation are not available so this is not necessary any more.

File: grunt/run.js [hunk 2]
```diff
@@ -13,5 +12,8 @@ module.exports = {
     },
     build_library: {
         exec: 'npm run build:library'
+    },
+    build_documentation_production: {
+        exec: 'npm run build:documentation:prod'
```
[the remark this answers is not in the document]
Comment (ashley-hunter) [reply grunt/run.js]: Unfortunately, grunt webpack was throwing errors when I tried to use it, I could not figure out why as it works fine when running from the command line directly or from an npm script

[the remark this answers is not in the document]
Comment (ashley-hunter) [reply grunt/run.js]: Unfortunately the updated version does not fix the issue

File: docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (donna-marijne) [reply docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html]: Understood, perhaps a slight rewording here: "This is only required when using Angular 5"

**Pull request merged**